Migration guide from track1 to track2 (#22879)
Azure/azure-sdk-for-net: sdk/resourcemanager/Azure.ResourceManager/README.md

to a specific resource in Azure.

### Structured Resource Identifier

> Resource IDs contains useful information about the resource itself, but they are plain strings that have to be parsed. Instead of implementing your own parsing logic, you can implicitly cast a resource identifier string into an object which will do the parsing for you.

Reviewer: instead of encouraging implicit cast encourage the constructor that takes a string

new ResourceIdentifer("myid");

> #### Casting a plain resource ID into a `ResourceIdentifier`

Reviewer: use the ResourceIdentifier ctor here

to a specific resource in Azure.

### Structured Resource Identifier

> Resource IDs contains useful information about the resource itself, but they are plain strings that have to be parsed. Instead of implementing your own parsing logic, you can use a `ResourceIdentifier` object which will do the parsing for you: `new ResourceIdentifer("myid");`.

Reviewer 2: ```suggestion
Resource IDs contain useful information about the resource itself, but they are plain strings that have to be parsed. Instead of implementing your own parsing logic, you can use a `ResourceIdentifier` object which will do the parsing for you: `new ResourceIdentifer("myid");`.
```

Console.WriteLine($"Vnet: {id.Parent.Name}");
Console.WriteLine($"Subnet: {id.Name}");
```

> However, keep in mind that some of those properties could be null. You can usually tell by the id string itself which type a resource ID is, but if you are unsure, use the Try methods to retrieve the values as it's shown below:

Reviewer: I would say you can check if the properties are null OR use the try methods

again use the ctor for ResourceIdentifier

AvailabilitySet availabilitySet = await availabilitySetOperations.GetAsync();
```

### `GetIfExists()` and `CheckIfExists` convenience methods

> If you are not sure if a resource you want to get exists, or you just want to check if it exists, you can use `GetIfExists()` or `CheckIfExists()` methods, which can be invoke from any [Resource]Container class.

Reviewer 2: ```suggestion
If you are not sure if a resource you want to get exists, or you just want to check if it exists, you can use `GetIfExists()` or `CheckIfExists()` methods, which can be invoked from any [Resource]Container class.
```

> `GetIfExists()` and `GetIfExistsAsync()` are going to return a null object if the specified resource name or id does not exists. On the other hand, `CheckIfExists()` and `CheckIfExistsAsync()` is going to return a boolean, depending if the specified resource exists.

Reviewer 2: ```suggestion
`GetIfExists()` and `GetIfExistsAsync()` return a null object if the specified resource does not exist. On the other hand, `CheckIfExists()` and `CheckIfExistsAsync()` return ```false``` if the specified resource does not exist.
```

// With the container, we can create a new resource group with an specific name
string rgName = "myRgName";
Location location = Location.WestUS2;

> var rgData = new ResourceGroupData(location);

Reviewer: don't use var
